### ISSUE TITLE
UTF-8 BOM marker causes incorrect .eslintignore parsing

### ISSUE THREAD

**Issue by ssg** [eslint/eslint#11777]
**Tell us about your environment**
Windows 10 x64 1803

* **ESLint Version:**
v5.16.0

* **Node Version:**
v10.15.3

* **npm Version:**
6.4.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

The issue happens on all configurations.
</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
npx eslint src
```

**What did you expect to happen?**
.eslintignore to be honored.

**What actually happened? Please include the actual, raw output from ESLint.**
JS files still get analyzed despite that they match .eslintignore. Removing UTF-8 BOM marker from .eslintignore file fixes the issue. The applicability is broad as UTF-8 BOM markers are default on Visual Studio which means that creating the .eslintignore file in Visual Studio will always cause this issue.

I think this is a bug in Node.js FS package but apparently it's very controversial to fix, so this seems to be in need of a fix here.

**Are you willing to submit a pull request to fix this bug?**
I'm not familiar with the code, but why not?
